src/Symfony/Component/HttpClient/Response/MockResponse.php

```diff
@@ -297,5 +297,9 @@ private static function readResponse(self $response, array $options, ResponseInt
         if ($dlSize && $offset !== $dlSize) {
             throw new TransportException(sprintf('Transfer closed with %d bytes remaining to read.', $dlSize - $offset));
         }
+
+        if (null !== $response->info['error']) {
```

Comment:
we don't need to check if the key error exist ? 
```suggestion
        if (isset ($response->info['error']) && null !== $response->info['error']) {
```

Reply:
The `error` key must exist because it was merged from `ResponseInterface::getInfo()` above and `error` is a mandatory info.